aquacomputer_d5next.c

```diff
@@ -1639,28 +1639,32 @@ static int aqc_write(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 					if (ctrl_mode > 2)
 						return -EINVAL;
 				}
-				break;
-			default:
-				return -EOPNOTSUPP;
-			}
 
-			if (val == 0) {
-				/* Set the fan to 100% as we don't control it anymore */
-				ret =
-				    aqc_set_ctrl_val(priv,
-						     priv->fan_ctrl_offsets[channel] +
-						     AQC_FAN_CTRL_PWM_OFFSET,
-						     aqc_pwm_to_percent(255), AQC_BE16);
+				if (val == 0) {
+					/* Set the fan to 100% as we don't control it anymore */
+					ctrl_values_offsets[1] = priv->fan_ctrl_offsets[channel] +
+					    AQC_FAN_CTRL_PWM_OFFSET;
+					ctrl_values[1] = aqc_pwm_to_percent(255);
+					ctrl_values_types[1] = AQC_BE16;
+					num_ctrl_values = 2;
```

Comment:
In the [writing commit](https://github.com/aleksamagicka/aquacomputer_d5next-hwmon/pull/91/commits/e576dd8dc9fc6b209370791827b1fc465e867be9#diff-b0eeb47bb02c97a51a04cecf13c9b19ee9a4bbea5c45ad9ceee2afaba29dc51fR1679) you set `num_ctrl_values` as the first thing in the if body, and here as the last. Either way is fine by me, but unify the placement across the commits (so either first or last).